Library/Homebrew/api.rb

```diff
@@ -40,18 +44,28 @@ def fetch(endpoint)
     sig { params(endpoint: String, target: Pathname).returns(Hash) }
     def fetch_json_api_file(endpoint, target:)
       retry_count = 0
-      url = "#{API_DOMAIN}/#{endpoint}"
-      curl_args = %W[--compressed --silent #{url}]
+      url = "#{Homebrew::EnvConfig.api_domain}/#{endpoint}"
+      default_url = "#{HOMEBREW_API_DEFAULT_DOMAIN}/#{endpoint}"
+      curl_args = %w[--compressed --silent]
       curl_args.prepend("--time-cond", target) if target.exist? && !target.empty?
 
       begin
         # Disable retries here, we handle them ourselves below.
-        Utils::Curl.curl_download(*curl_args, to: target, max_time: JSON_API_MAX_TIME, retries: 0)
+        Utils::Curl.curl_download(*curl_args, url, to: target, max_time: JSON_API_MAX_TIME, retries: 0)
```

Comment:
This command prints a "404 error" to the stderr. Because `--show-error` is always added to the `curl` options.

https://github.com/Homebrew/brew/blob/791022dc2195f5caf819a91552014ec9546b20b8/Library/Homebrew/utils/curl.rb#L79

Reply:
@XuehaiPan We could consider making that conditional if desired.

Reply:
@MikeMcQuaid I pushed a new commit to make it optional.